sklearn/naive_bayes.py

```diff
@@ -39,6 +39,83 @@
 class _BaseNB(ClassifierMixin, BaseEstimator, metaclass=ABCMeta):
     """Abstract base class for naive Bayes estimators"""
 
+    _parameter_constraints: dict = {
+        "priors": ["array-like", StrOptions({"uniform", "empirical"}), None],
+    }
+
+    def __init__(self, *, priors=None):
+        self.priors = priors
+
+    def _validate_priors(self):
```

Comment:
This is the main changes to handle smoothly the deprecation in the discrete naive Bayes.